pkg/cmd/infra/router/template.go

```diff
@@ -225,6 +232,37 @@ func NewCommandTemplateRouter(name string) *cobra.Command {
 	return cmd
 }
 
+// httpHeaderNameRE matches valid HTTP header names.
+var httpHeaderNameRE *regexp.Regexp = regexp.MustCompile(`^[\x21\x23-\x27\x2a\x2b\x2d\x2e\x30-\x39\x41-\x5a\x5e-\x7a\x7c\x7e]+$`)
```

Comment:
Mostly because i'm curious, what specifically does this reg ex match? How did you come up with it?

Reply:
It matches valid header names.  I constructed it by painstakingly following the spec:

       message-header = field-name ":" [ field-value ]
       field-name     = token

https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

       token          = 1*<any CHAR except CTLs or separators>
       CHAR           = <any US-ASCII character (octets 0 - 127)>
       CTL            = <any US-ASCII control character
                        (octets 0 - 31) and DEL (127)>
       separators     = "(" | ")" | "<" | ">" | "@"
                      | "," | ";" | ":" | "\" | <">
                      | "/" | "[" | "]" | "?" | "="
                      | "{" | "}" | SP | HT
       SP             = <US-ASCII SP, space (32)>
       HT             = <US-ASCII HT, horizontal-tab (9)>

https://www.w3.org/Protocols/rfc2616/rfc2616-sec2.html#sec2.2

Reply:
Great. Would you mind enhancing the `httpHeaderNameRE` comment by mentioning RFC 2616, like you did in https://github.com/openshift/enhancements/pull/370, just to make sure the question doesn't pop up again?

Reply:
Done.